### PR TITLE
Fix an issue with the new Pillow update

### PR DESCRIPTION
**Pull request by njooma**
Pillow [pulled some nonsense](https://github.com/python-pillow/Pillow/pull/7307) by making a breaking change in a minor version. Oops on them. Panic! at the Robot Factory.

### CODE REVIEW

File: src/viam/media/viam_rgba_plugin.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def _open(self):
         width = int.from_bytes(header[4:8], "big")
         height = int.from_bytes(header[8:12], "big")
         self._size = width, height
-        self.mode = "RGBA"
+        self._mode = "RGBA"
```
[the remark this answers is not in the document]
Comment (njooma) [reply src/viam/media/viam_rgba_plugin.py:58]: I believe that's for existing images, but not for creating our own image decoder. The [documentation](https://pillow.readthedocs.io/en/stable/handbook/writing-your-own-image-plugin.html) states
>An image plugin should contain a format handler derived from the [PIL.ImageFile.ImageFile](https://pillow.readthedocs.io/en/stable/reference/ImageFile.html#PIL.ImageFile.ImageFile) base class. This class should provide an _open method, which reads the file header and set at least the internal _size and _mode attributes so that [mode](https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.mode) and [size](https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.size) are populated.

so we should be safe setting these private attrs here. (famous last words, given this PR)

File: examples/server/v1/components.py [hunk 2]
```diff
@@ -338,6 +338,9 @@ async def model_attributes(self) -> Board.Attributes:
     async def set_power_mode(self, **kwargs):
         raise NotImplementedError()
 
+    async def write_analog(self, pin: str, value: int, *, timeout: float | None = None, **kwargs):
+        raise NotImplementedError()
```
Comment (benjirewis) [inline examples/server/v1/components.py:342]: [q] how is this related to the `mode` change? just a drive-by?

**Review by benjirewis — APPROVED** [review]
